### DISCUSSION TITLE
How do i generate report html/xml while the test is running?

### DISCUSSION THREAD

**Discussion by ws1088** [pytest-dev/pytest#8390] [Q&A]
Currently I *think* I see that html or junit xml reports are generated after all tests are completed.

**Comment by RonnyPfannschmidt** [comment]
correct, currently there is no live gen as the junit-xml formats want to include global stats on the first starting xml element and nobody made the effort of creating a serialize mechanism that can live-update those in a sensible manner

**Reply by ws1088** [reply]
thanks. but i guess i can write a plugin to catch the test pass/fail/stdout/stderr as they come in and store it somewhere for on-the-fly display?

**Reply by RonnyPfannschmidt** [reply]
You can

**Reply by xhuw** [reply]
@ws1088  did you have any luck with this? any chance I will find your plugin on pypi? Thanks

**Reply by The-Compiler** [reply]
For what it's worth, [pytest-reportlog](https://github.com/pytest-dev/pytest-reportlog) does stream its JSON lines, and also contains the output (at least for failing tests).
